src/client/dfuse/il/int_posix.c

```diff
@@ -729,6 +793,20 @@ call_daos_init(int fd)
 	return rcb;
 }
 
+static void
+child_hdlr(void)
+{
+	int rc;
+
+	daos_dti_reset();
+	ioil_eqh = DAOS_HDL_INVAL;
+	rc = daos_eq_create(&ioil_eqh);
+	if (rc)
+		DFUSE_LOG_WARNING("daos_eq_create() failed: "DF_RC, DP_RC(rc));
+	else
+		ioil_iog.iog_main_eqh = ioil_eqh;
```

Comment:
does fio actually do I/O in the forked child without subsequent exec?  Seems like a lot of stuff can go wrong in that scenario.   Like the log message above.  Do we need to reinitialize the logging?

Reply:
yes. actually those forked processes don't get to finalize, so the stat / logging don't even show up / get printed.
anyway for that scenario, my intent is to just get FIO to work and perform well, and not worry about stats or leaked contexts in this case. we really cannot / should not claim we support that scenario today.

Reply:
As @wiliamhuang mentioned, we can do more  to have this fio case work better by intercepting _exit() and do the finalize and printing of stats, etc.
i don't want to do that in this PR though and we can create a ticket and fix that in a future PR.

Reply:
Have you tried with D_LOG_FILE?  I suspect it has issues with current patch

Reply:
no the parent doesn't exit. the children just do their IO job and _exit()
we don't need to re-init logging because logging for the children don't get printed anyway in the FIO case.

i will reiterate again that im not interested in implementing support for fork() in this PR. this PR is at least better than what we end up with today (a segfault in libfabric).

what is the issue with D_LOG_FILE?

Reply:
sorry i confused the logging that you are talking about with the stats of the IL. 
but im not getting any logging errors in my case and D_LOG_FILE works fine.

Reply:
i just tried a quick run setting D_LOG_FILE and debug mask and it's fine.
as for what to recommend for the user, we should still not recommend fork as beyond this FIO use case, because any usage of dfs metadata calls or any daos calls with NULL event, we get back to the libfabric segfault. so this is not something we can support today anyway, but just a workaround for FIO testing.

Reply:
Ok, it does just seem to work which is surprising

jeffolivier@jeffolivier1:~/community/daos$ cat foo.c
#include <unistd.h>
#include <gurt/debug.h>

int main(int argc, char **argv)
{
	int rc = d_log_init_adv("DAOS", "/tmp/foo.log", 0, DLOG_INFO, DLOG_CRIT, NULL);
	int pid = getpid();
	int child_pid = fork();
	if (child_pid == 0) {
		D_INFO("Log from child\n");
	}
	D_INFO("Log from both\n");
	d_log_fini();
}

effolivier@jeffolivier1:~/community/daos$ cat /tmp/foo.log 
09/11-21:52:48.29 jeffolivier1 INFO foo.c:15 main() Log from both
09/11-21:52:48.29 jeffolivier1 INFO foo.c:12 main() Log from child
09/11-21:52:48.29 jeffolivier1 INFO foo.c:15 main() Log from both